### ISSUE TITLE
About A3C

### ISSUE THREAD

**Issue by Tord-Zhang** [Kaixhin/Atari#66]
I just read the code and find that only policy network(critic) is trained,while there is no value work(actor).So I think it is a policy based RL network rather than a actor-critic network?Maybe I don't understand A3C right or maybe I just don't read the code carefully enough.

**Comment by Kaixhin** [comment]
What is denoted `policyNet` in the code contains both the value prediction and policy with a shared body. The construction of this can be found in the [model code](https://github.com/Kaixhin/Atari/blob/master/Model.lua#L190-L208). So when using the network, [both the value and the policy are extracted](https://github.com/Kaixhin/Atari/blob/master/async/A3CAgent.lua#L87). In the training code, the [value function target](https://github.com/Kaixhin/Atari/blob/master/async/A3CAgent.lua#L90) and the [policy gradients](https://github.com/Kaixhin/Atari/blob/master/async/A3CAgent.lua#L92-L95) are both calculated and then [backpropagated](https://github.com/Kaixhin/Atari/blob/master/async/A3CAgent.lua#L102).
